### PR TITLE
Nut80

### PR DESCRIPTION
**Pull request by prakharcode**
@helgee the error is coming in `mod2pi` corresponding to `fmod`. Tried a bunch of things didn't work.

### CODE REVIEW

File: src/AstroBase.jl [hunk 1]
```diff
@@ -115,4 +115,78 @@ function tio_locator(jd1, jd2)
...
+Returns an angle(radians) in the range [-π, π] for a given angle(radians).
+
+# Example
+
+```jldoctest
+julia> normalize_pi_angle(21)
+2.1504440784612413
+```
+"""
+function normalize_pi_angle(a)
```
Comment (giordano) [inline src/AstroBase.jl]: This is just
```julia
normalize_pi_angle(a) = rem2pi(a, RoundNearest)
```

File: src/AstroBase.jl [hunk 2]
```diff
@@ -115,4 +115,78 @@ function tio_locator(jd1, jd2)
...
+    for j in NT:-1:1
+        arg = multiples_of_arguments_and_coefficients[j][1]  * mean_longitude_moon_minus_mean_longitude_moon_perigee
+        + multiples_of_arguments_and_coefficients[j][2] * mean_longitude_sun_minus_mean_longitude_sun_perigee
+        + multiples_of_arguments_and_coefficients[j][3]  * mean_longitude_moon_minus_mean_longitude_moon_node
+        + multiples_of_arguments_and_coefficients[j][4]  * mean_elongation_moon_from_sun
+        + multiples_of_arguments_and_coefficients[j][5] * mean_ascending_node_lunar_orbit_ecliptic_measured_mean_equinox_date
+
+        s = multiples_of_arguments_and_coefficients[j][6] + multiples_of_arguments_and_coefficients[j][7] * t
+        c = multiples_of_arguments_and_coefficients[j][8] + multiples_of_arguments_and_coefficients[j][9] * t
+        if s != 0.0 dp += s * sin(arg) end
```
Comment (giordano) [inline src/AstroBase.jl]: Personally, I don't really like this one-line `if`. I'd do:
```julia
sinarg, cosarg = sincos(arg)
iszero(s) || ds += s * sinarg
iszero(c) || de += c * cosarg
```
I don't know what @helgee thinks about it, anyway I recommend using `sincos` and `iszero`.

Comment (helgee) [reply src/AstroBase.jl]: I concur. As I have suggested in #25 we should move to Julia 0.7 which makes `sincos` available.

File: src/AstroBase.jl [hunk 3]
```diff
@@ -115,4 +115,78 @@ function tio_locator(jd1, jd2)
...
+# Example
+
+```jldoctest
+julia> normalize_pi_angle(21)
+2.1504440784612413
+```
+"""
+function normalize_pi_angle(a)
+    w = mod(a, 2pi)
+    if abs(w) >= pi w -= (a < 0.0? -abs(w): abs(w)) end
```
Comment (helgee) [inline src/AstroBase.jl]: Apart from @giordano's comment above. Oneliners like this are a no-go. There are several branches and a mutation of a variable on one line. This is like asking for trouble...

File: src/AstroBase.jl [hunk 4]
```diff
@@ -19,14 +19,17 @@ export tio_locator,
...
     obliquity_of_ecliptic_06,
     mean_obliquity_of_ecliptic,
-    precession_fukushima_williams06
+    precession_fukushima_williams06,
+    nutation
 
 const J2000 = 2451545.0
 const DAYS_PER_CENTURY = 36525.0
 const ARCSECONDS_IN_CIRCLE = 1296000.0
+const U2R = deg2rad(1/1e4 * (1/3600))
```
Comment (giordano) [inline src/AstroBase.jl]: Why not `sec2rad(1e-4)`?

File: src/AstroBase.jl [hunk 5]
```diff
@@ -485,4 +489,60 @@ function xy06(jd1, jd2)
...
+```jldoctest
+julia> nutation(2.4578265e6, 0.30434616919175345)
+(-3.7565297299394694e-5, -3.665617105048724e-5
+```
+"""
+function nutation(jd1, jd2)
+
+    t = ((jd1 - J2000) + jd2) / DAYS_PER_CENTURY
+
+    mean_longitude_moon_minus_mean_longitude_moon_perigee = rem2pi(
```
Comment (giordano) [inline src/AstroBase.jl:703]: So you decided to discard `normalize_pi_angle` function?  It's fine with me as its definition is quite simple, but I think it wouldn't hurt either.

Comment (prakharcode) [reply src/AstroBase.jl:703]: Yes, I thought it was causing verbosity. Should I reimplement it?

Comment (giordano) [reply src/AstroBase.jl:703]: It's fine with me to keep this way, but let @helgee decide :sweat_smile:

Comment (prakharcode) [reply src/AstroBase.jl:703]: Sure!

Comment (helgee) [reply src/AstroBase.jl:703]: Let's keep it this way. 👍

File: src/constants.jl [hunk 6]
```diff
@@ -0,0 +1,130 @@
+const multiples_of_arguments_and_coefficients = [
```
Comment (helgee) [inline src/constants.jl]: This should probably be a tuple of tuples.

**Pull request merged**